tests/test_connector.py

```diff
@@ -200,6 +200,24 @@ def test_get_session_relogin(self, requests_mock):
         assert r.status == 200
         assert r.json is None
 
+    def test_get_session_relogin_delete_bad_token_header(self, requests_mock):
+        requests_mock.get("https://demo.dev/data", [
+            dict(status_code=401), dict(status_code=200),
+        ])
+        requests_mock.post(
+            "https://demo.dev/sessions", status_code=201,
+            headers={"X-Auth-Token": "123", "Location": "/sessions/3"},
+        )
+        requests_mock.post(
+            "https://demo.dev/sessions", status_code=401,
+            request_headers={"x-auth-token": "abc"},
```

Comment:
Am I missing something or is this second POST to /sessions supposed to be without the x-auth-token header? I thought this is what the patch is all about?

Reply:
It turns out that it is quite hard to test for "this request should not have this header set". This is why this test is a bit contrived. I will add a comment describing the mechanics, but if you have any other suggestion on how to efficiently and adequately test this, I would greatly appreciate it.